test/integration/integration_test.go

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package integration
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/AzureAD/microsoft-authentication-library-for-go/msal"
+)
+
+// HTTP Helper
+func httpRequest(url string, query map[string]string, accessToken string) ([]byte, error) {
```

Comment:
The URL package includes a value used for queries: https://golang.org/pkg/net/url/#Values

If you use it instead, you can simply do:
request.URL.RawQuery = query.Encode()